server/schedulers/balance_leader.go

```diff
@@ -71,15 +80,105 @@ func init() {
 }
 
 type balanceLeaderSchedulerConfig struct {
-	Name   string          `json:"name"`
-	Ranges []core.KeyRange `json:"ranges"`
-	Batch  int             `json:"batch"`
+	mu      sync.RWMutex
+	storage endpoint.ConfigStorage
+	Name    string          `json:"name"`
+	Ranges  []core.KeyRange `json:"ranges"`
+	Batch   int             `json:"batch"`
+}
+
+func (conf *balanceLeaderSchedulerConfig) Update(data []byte) (int, interface{}) {
+	conf.mu.Lock()
+	defer conf.mu.Unlock()
+
+	oldc, _ := json.Marshal(conf)
+
+	if err := json.Unmarshal(data, conf); err != nil {
+		return http.StatusInternalServerError, err.Error()
+	}
+	newc, _ := json.Marshal(conf)
+	if !bytes.Equal(oldc, newc) {
+		conf.persistLocked()
+		return http.StatusOK, "success"
+	}
+	m := make(map[string]interface{})
+	if err := json.Unmarshal(data, &m); err != nil {
+		return http.StatusInternalServerError, err.Error()
+	}
+
+	t := reflect.TypeOf(conf).Elem()
+	for i := 0; i < t.NumField(); i++ {
+		jsonTag := t.Field(i).Tag.Get("json")
+		if i := strings.Index(jsonTag, ","); i != -1 { // trim 'foobar,string' to 'foobar'
+			jsonTag = jsonTag[:i]
+		}
+		if _, ok := m[jsonTag]; ok {
+			return http.StatusOK, "no changed"
```

Comment:
```suggestion
			return http.StatusOK, http.StatusText(http.StatusOK)
```
the status text should be capitalized.